plugins/woocommerce-admin/client/two-column-tasks/sectioned-task-list.tsx

```diff
@@ -0,0 +1,286 @@
+		return {
+			profileItems: getProfileItems(),
+		};
+	} );
+	const { hideTaskList } = useDispatch( ONBOARDING_STORE_NAME );
+	const [ openPanel, setOpenPanel ] = useState< string | null >(
+		sections?.find( ( section ) => ! section.isComplete )?.id || null
+	);
+
+	const prevQueryRef = useRef( query );
+
+	const nowTimestamp = Date.now();
+	const visibleTasks = tasks.filter(
+		( task ) =>
+			! task.isDismissed &&
+			( ! task.isSnoozed || task.snoozedUntil < nowTimestamp )
+	);
+
+	const recordTaskListView = () => {
+		if ( query.task ) {
+			return;
+		}
+
+		recordEvent( `${ eventName }_view`, {
+			number_tasks: visibleTasks.length,
+			store_connected: profileItems.wccom_connected,
+		} );
+	};
+
+	useEffect( () => {
+		recordTaskListView();
+	}, [] );
+
+	useEffect( () => {
+		const { task: prevTask } = prevQueryRef.current;
+		const { task } = query;
+
+		if ( prevTask !== task ) {
+			window.document.documentElement.scrollTop = 0;
+			prevQueryRef.current = query;
+		}
+	}, [ query ] );
+
+	const onDismissTask = ( taskId: string, onDismiss?: () => void ) => {
+		dismissTask( taskId );
+		createNotice( 'success', __( 'Task dismissed' ), {
+			actions: [
+				{
+					label: __( 'Undo', 'woocommerce-admin' ),
+					onClick: () => undoDismissTask( taskId ),
+				},
+			],
+		} );
+
+		if ( onDismiss ) {
+			onDismiss();
+		}
+	};
+
+	const hideTasks = () => {
+		hideTaskList( id );
+	};
+
+	const keepTasks = () => {
+		const updateOptionsParams = {
+			woocommerce_task_list_keep_completed: 'yes',
+		};
+
+		updateOptions( {
+			...updateOptionsParams,
+		} );
+	};
+
+	let selectedHeaderCard = visibleTasks.find(
+		( listTask ) => listTask.isComplete === false
+	);
+
+	// If nothing is selected, default to the last task since everything is completed.
+	if ( ! selectedHeaderCard ) {
+		selectedHeaderCard = visibleTasks[ visibleTasks.length - 1 ];
+	}
+
+	const trackClick = ( task: TaskType ) => {
+		recordEvent( `${ eventName }_click`, {
+			task_name: task.id,
+		} );
+	};
+
+	const goToTask = ( task: TaskType ) => {
+		trackClick( task );
+		updateQueryString( { task: task.id } );
+	};
+
+	const onTaskSelected = ( task: TaskType ) => {
+		goToTask( task );
+	};
+
+	const getSectionTasks = ( sectionTaskIds: string[] ) => {
+		return visibleTasks.filter( ( task ) =>
+			sectionTaskIds.includes( task.id )
+		);
+	};
+
+	const getPanelTitle = ( section: TaskListSection ) => {
+		if ( openPanel === section.id ) {
+			return (
+				<div className="wooocommerce-task-card__header-container">
+					<div className="wooocommerce-task-card__header">
+						<SectionHeader { ...section } />
+					</div>
+				</div>
+			);
+		}
+		const completedTasksCount = tasks.filter(
+			( task ) => ! task.isComplete && section.tasks.includes( task.id )
+		).length;
+		return (
+			<>
+				<Text variant="title.small" size="20" lineHeight="28px">
+					{ section.title }
+				</Text>
+				{ ! section.isComplete && (
+					<Badge count={ completedTasksCount } />
```

Comment:
After completing a section that has 1 task left and navigating back to the homescreen without page refresh (e.g., visit "Get ready to sell" with just the shipping task left, complete it, and then click "Home" without refreshing), the badge gets updated to `0` instead of a completed check mark:
<img width="414" alt="Screen Shot 2022-04-12 at 3 02 14 PM" src="https://user-images.githubusercontent.com/10561050/163035289-dfa82729-f370-4d5e-b82e-937de7742f5a.png">

Reply:
That is likely the delay between the optimistic task completion update and when it does the full task list update.

I updated the logic to show the check mark if uncompleted count is zero.